src/proc.cpp

```diff
@@ -681,8 +684,9 @@ void proc_update_jiffies() {
 ///
 /// \param j the job to test
 ///
-/// \return 1 if buffers were available, zero otherwise
-static int select_try(job_t *j) {
+/// \return Tristate::yes() if buffers were ready, Tristate::no() if no buffers were ready,
+/// and Tristate::err() on error.
+static Tristate select_try(job_t *j) {
```

Comment:
Rather than this Tristate abstraction which IMO won't prove useful enough to justify, let's use an `enum class` here:

    enum class select_return_t {
          some_readable,   // some fd was readable
          none_readable,  // no fd was readable
          no_fds,  // no fds were present in the IO chain
    };

(probably you can think of better value names).

I think this will be more explicit.